osquery/tables/networking/posix/lldp_neighbors.cpp

```diff
@@ -0,0 +1,421 @@
+  }
+}
+
+void LLDPNeighbor::getChassis() {
+  row_["rid"] = getAtomStr(chassis_, lldpctl_k_chassis_index);
+  row_["chassis_id"] = getAtomStr(chassis_, lldpctl_k_chassis_id);
+  row_["chassis_id_type"] = getAtomStr(chassis_, lldpctl_k_chassis_id_subtype);
+  row_["chassis_sysname"] = getAtomStr(chassis_, lldpctl_k_chassis_name);
+  row_["chassis_sys_description"] =
+      getAtomStr(chassis_, lldpctl_k_chassis_descr);
+
+  getChasisCapability(LLDP_CAP_BRIDGE,
+                      "chassis_bridge_capability_available",
+                      "chassis_bridge_capability_enabled");
+  getChasisCapability(LLDP_CAP_ROUTER,
+                      "chassis_router_capability_available",
+                      "chassis_router_capability_enabled");
+  getChasisCapability(LLDP_CAP_WLAN,
+                      "chassis_wlan_capability_available",
+                      "chassis_wlan_capability_enabled");
+  getChasisCapability(LLDP_CAP_REPEATER,
+                      "chassis_repeater_capability_available",
+                      "chassis_repeater_capability_enabled");
+  getChasisCapability(LLDP_CAP_TELEPHONE,
+                      "chassis_tel_capability_available",
+                      "chassis_tel_capability_enabled");
+  getChasisCapability(LLDP_CAP_DOCSIS,
+                      "chassis_docsis_capability_available",
+                      "chassis_docsis_capability_enabled");
+  getChasisCapability(LLDP_CAP_STATION,
+                      "chassis_station_capability_available",
+                      "chassis_station_capability_enabled");
+  getChasisCapability(LLDP_CAP_OTHER,
+                      "chassis_other_capability_available",
+                      "chassis_other_capability_enabled");
+
+  lldpctl_atom_t* mgmts = lldpctl_atom_get(chassis_, lldpctl_k_chassis_mgmt);
+  row_["chassis_mgmt_ips"] = commaDelimitedStr(mgmts, lldpctl_k_mgmt_ip);
+  lldpctl_atom_dec_ref(mgmts);
+}
+
+/**
+ * @brief getNeighbor retreives all LLDP information of the given neighbor port
+ * and chassis.
+ *
+ * @return filled osquery::Row of lldp information for a given interface
+ *
+ */
+Row& LLDPNeighbor::getNeighbor() {
+  getChassis();
+  getPort();
+  getVLAN();
+  getMED();
+  getMEDCap();
+  getPIDs();
+  getPPVIDs();
+
+  return row_;
+}
+
+QueryData genLLDPNeighbors(QueryContext& context) {
+  QueryData rows;
+
+  lldpctl_conn_t* conn(lldpctl_new(nullptr, nullptr, nullptr));
+  lldpctl_error_t err = lldpctl_last_error(conn);
+  if (err != LLDPCTL_NO_ERROR) {
+    LOG(ERROR) << "could not initiate new lldpd connection: "
+               << lldpctl_strerror(err);
+    return rows;
+  }
+
+  lldpctl_atom_t* interfaces = lldpctl_get_interfaces(conn);
+  err = lldpctl_last_error(conn);
+  if (err != LLDPCTL_NO_ERROR) {
+    LOG(WARNING) << "could not connect to lldpd (hint: you might need to "
+                    "install lldpd v0.9.X or run in sudo): "
+                 << lldpctl_strerror(err);
+    return rows;
```

Comment:
Yeah it's been updated. `conn` is now a unique_ptr now with a custom deleter that handles the de-reffing.